src/share/import.js

```diff
@@ -0,0 +1,77 @@
+import fs from '../lib/fs';
+import Buffer from '../lib/buffer';
+import Path from '../lib/path';
+import WebTorrent from 'webtorrent';
+const sh = new fs.Shell();
+
+document.getElementById('torrentId').value = window.location.search.substring(
+    1
+);
+document.getElementById('btnDownload').addEventListener('click', startDownload);
+
+const downloadAllTorrentFiles = torrent =>
+    Promise.all(
+        torrent.files.map(
+            file =>
+                new Promise((resolve, reject) => {
+                    console.log('file: ', file);
+                    var filename = Path.join('/', file.path);
+                    file.getBuffer((err, buffer) => {
+                        if (err) {
+                            writeError();
+                            return reject(err);
+                        }
+                        sh.mkdirp(Path.dirname(filename), err => {
+                            if (err) {
+                                console.error(
+                                    'Error writing file, error was ',
+                                    err,
+                                    filename
+                                );
+                                writeError();
+                                return reject(err);
+                            }
+                            fs.writeFile(filename, new Buffer(buffer), err => {
+                                if (err) {
+                                    console.error(
+                                        'Error writing file, error was ',
+                                        err,
+                                        filename
+                                    );
+                                    writeError();
+                                    return reject(err);
+                                }
+                                console.log('Wrote to ', filename);

```

Comment:
We probably don't need this.  Maybe later we should do some kind of progress info in the page so people can see that things are happening.